app/models/event_stream.rb

```diff
@@ -54,20 +56,21 @@ def self.event_groups
   end
 
   def self.group_and_level(event_type)
-    group, v = event_groups.find { |_k, v| v[:critical].include?(event_type) || v[:detail].include?(event_type) }
-    if group.nil?
-      group, level = :other, :detail
-    else
-      level = v[:detail].include?(event_type) ? :detail : :critical
+    level = :detail # the level is detail as default
+    group = event_groups.find do |_k, value|
+      lvl_found = GROUP_LEVELS.detect { |lvl| value[lvl]&.include?(event_type) }
+      level = lvl_found if lvl_found # update the level variable if found the current level.
+      lvl_found # the level found or nil
```

Comment:
How about this?
```
  def self.group_and_level(event_type)
    group, value = event_groups.detect { |_k, v| GROUP_LEVELS.any? { |g| v[g]&.include?(event_type) } }
    if group.nil?
      group, level = :other, :detail
    else
      level = GROUP_LEVELS.detect { |g| value[g].include?(event_type) }
    end
    return group, level
  end
```

Reply:
@lfu This works, but I think that is more complex, cause you need to do one more iteration. So I prefer the another form, but thank you.

Reply:
It can be simplified as:
```
  def self.group_and_level(event_type)
    level = :detail # the level is detail as default
    group, _ = event_groups.find do |_k, value|
      GROUP_LEVELS.detect { |lvl| value[lvl]&.include?(event_type) }.tap do |level_found|
        level = level_found if level_found
      end
    end

    group ||= :other
    return group, level
  end
```

Reply:
I liked this, thank you :smile: